src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,169 @@
+        return sc.nextLine();
+    }
+
+    /**
+     * Prints the error message of a DukeException.
+     *
+     * @param eMsg the error message of the DukeException.
+     */
+    public void showError(String eMsg) {
+        System.out.println(line
+                + eMsg + "\n"
+                + line);
+        showingString = eMsg + "\n";
+    }
+
+    private void printLine() {
+        System.out.print(line);
+    }
+
+    /**
+     * Prints out the list of tasks.
+     *
+     * @param taskList the list of tasks.
+     */
+    public void printList(List<Task> taskList) {
+        if (taskList.size() == 0) {
+            printLine();
+            System.out.println(" Oops, no task YET. Try to add one!");
+            printLine();
+            showingString = " Oops, no task YET. Try to add one!";
+            return;
+        }
+        StringBuffer finalString = new StringBuffer();
+        printLine();
+        finalString.append(" Here are the tasks in your list:\n");
+        for (int i = 0; i < taskList.size(); i++) {
+            finalString.append(" ").append(i + 1).append(". ").append(taskList.get(i)).append("\n");
+        }
+        System.out.println(finalString);
+        printLine();
+        showingString = finalString.toString();
+    }
+
+    /**
+     * Prints out the Done command executed on a certain task.
+     *
+     * @param task the task to be done.
+     */
+    public void printDone(Task task) {
+        String finalString = " Nice! I've marked this task as done: "
+                + "\n   " + task + "\n";
+        System.out.println(line
+                + finalString
+                + line);
+        showingString = finalString;
+    }
+
+    /**
+     * Prints out the Delete command executed on a certain task.
+     *
+     * @param task the task to be deleted.
+     * @param size the size of the task list after deletion.
+     */
+    public void printDelete(Task task, int size) {
+        String finalString = " Noted. I've removed this task: "
+                + "\n   " + task
+                + "\n Now you have " + size + " tasks in the list.\n";
+        System.out.println(line
+                + finalString
+                + line);
+        showingString = finalString;
+    }
+
+    /**
+     * Prints out the Add command.
+     *
+     * @param task the task to be added.
+     * @param size the size of the task list after addition.
+     */
+    public void printAdd(Task task, int size) {
+        String finalString = " Got it. I've added this task: \n"
+                + "   " + task
+                + "\n Now you have " + size + " tasks in the list.\n";
+        System.out.println(line
+                + finalString
+                + line);
+        showingString = finalString;
+    }
+
+    /**
+     * Prints out the list of tasks which was found by find command.
+     *
+     * @param findResult the list of tasks which was found.
+     */
+    public void printFind(List<Task> findResult) {
```

Comment:
findResult sounds a bit like verb I think. Probably we change to another name such as taskListResult?